packages/middleware-signing/src/middleware.spec.ts

```diff
@@ -2,166 +2,91 @@ import { HttpRequest } from "@aws-sdk/protocol-http";
-      signer: noOpSigner,
-      systemClockOffset,
-    } as any)(noOpNext, {} as any);
-    await signingHandler({
-      input: {},
-      request: new HttpRequest({
-        headers: {},
-      }),
-    });
-
-    const { calls } = (noOpNext as any).mock;
-    expect(calls.length).toBe(1);
-    expect(calls[0][0].request.headers.signed).toBe("true");
-    // Using greater than to ensure there are no timing issues
-    expect(calls[0][0].request.headers.signingDateTime).toBeGreaterThan(now + systemClockOffset - 1);
-  });
-
-  describe("update systemClockOffset if there is clockSkew", () => {
-    // Set up clockSkew as abs(newSystemClockOffset - systemClockOffset) > 300000
-    const clockSkewPresent = [
-      { current: 100000, new: 500000 },
-      { current: -100000, new: 250000 },
-      { current: 200000, new: -150000 },
-      { current: -100000, new: -450000 },
-    ];
-
-    clockSkewPresent.forEach((systemClockOffsetVal) => {
-      it(`current systemClockOffset: ${systemClockOffsetVal.current}, new systemClockOffset: ${systemClockOffsetVal.new}`, async () => {
-        expect.assertions(3);
-        const systemClockOffset = systemClockOffsetVal.current;
-        const newSystemClockOffset = systemClockOffsetVal.new;
-        const options = {
-          signer: noOpSigner,
-          systemClockOffset,
-        };
-        const signingHandler = awsAuthMiddleware(options as any)(noOpNext, {} as any);
-        noOpNext.mockReturnValue({
-          response: {
-            headers: {
-              date: new Date(Date.now() + newSystemClockOffset).toString(),
-            },
-          },
-        });
-
-        await signingHandler({
-          input: {},
-          request: new HttpRequest({
-            headers: {},
-          }),
-        });
-
-        const { calls } = (noOpNext as any).mock;
-        expect(calls.length).toBe(1);
-        expect(calls[0][0].request.headers.signed).toBe("true");
-        expect(options.systemClockOffset).not.toBe(systemClockOffset);
-      });
-    });
-  });
-
-  describe("do not update systemClockOffset if there is no clockSkew", () => {
-    // Do not set up clockSkew as abs(newSystemClockOffset - systemClockOffset) < 300000
-    const clockSkewNotPresent = [
-      { current: 100000, new: 250000 },
-      { current: -100000, new: 50000 },
-      { current: 50000, new: -150000 },
-      { current: -100000, new: -150000 },
-    ];
-
-    clockSkewNotPresent.forEach((systemClockOffsetVal) => {
-      it(`current systemClockOffset: ${systemClockOffsetVal.current}, new systemClockOffset: ${systemClockOffsetVal.new}`, async () => {
-        expect.assertions(3);
-        const systemClockOffset = systemClockOffsetVal.current;
-        const newSystemClockOffset = systemClockOffsetVal.new;
-        const options = {
-          signer: noOpSigner,
-          systemClockOffset,
-        };
+  describe("should update systemClockOffset if date header is present", () => {
```

Comment:
Add a case when date header does not exist?

Reply:
The first unit test checks for absence of date header by verifying that `getUpdatedSystemClockOffset` is not called.

https://github.com/trivikr/aws-sdk-js-v3/blob/13fbdb3548e1d3fb5632157b0abf0d0483c40d5d/packages/middleware-signing/src/middleware.spec.ts#L52-L59